### ISSUE TITLE
Small quality of life UI tweaks

### ISSUE THREAD

**Issue by darsain** [buttercup/buttercup-desktop#536]
#### Copy field values by clicking on the whole value area, not the small hidden icon to the right.

and related:

#### Don't hide field action buttons, like the eye to view the password

Buttons popping into existence when moving mouse cursor is distracting. I'm also not aware of them until I happen to hover over one, creating an illusion of missing features for new users.

Also, trying to click one means I have to aim for an invisible target, so you aim for the row, and then as the button pops into existence adjust your aim for the button, making sure you move properly horizontally, because hovering out of the field will hide it again. Isn't it easier to just aim straight for the button?

By making value field itself the copy field button, you've removed most of the current buttons, and are left only with the eye button to show passwords, so this will cause no UI clutter at all, but make the UI nicer to visually navigate and use.

#### Remove the "Custom fields" section divider

This divider servers no practical purpose, and it arbitrarily divides data of the same item for no semantic meaning to the user. It only informs me how are the fields implemented in the backend, which has no value to me. And it takes valuable UI space. And looks bad having the overall nice table of data divided like that :)

**Comment by perry-mitchell** [comment]
Hi @darsain - Thanks for going through all of this!

With regards to the copying fields by clicking, I'm not sure that this makes so much sense. For instance, I frequently copy only a portion of the field value. Having an auto-copy feature would hinder that and be less obvious from a UX perspective.

I'm also not completely convinced that hiding the buttons is so bad in terms of UX. It hides clutter (which is bad UX) and you instantly learn about them when you hover one field. If they were all visible the interface would look horrible as well, imho.

On the custom fields divider, I tend to somewhat agree with your sentiment there. We could just have them altogether, but I wouldn't make the username/password removable. In place of that, I would hope that we get #540 implemented so that some entry types that don't need those fields can hide them.

**Comment by darsain** [comment]
You can still select text no problem. Fields get copied on `click`. `mousedown->drag-to-select->mouseup` is not a click, or shouldn't be. So it's all an implementation detail. And I absolutely agree, going the Google Chrome route where you can't select text inside any anchor link is very bad.

Also, I thought click or double-click to copy is pretty much standard in password managers, no? It's also obvious by cursor changing to hand if you hover the field. It says there is an action upon clicking, and copy is the only action that makes sense in this context. At worst, you can add a little helpful hint in the bottom right corner that says "click values to copy them". I find the current situation of having to copy fields by clicking on that tiny and hidden icon very frustrating to be honest. 

And icons: when you have many, make a hamburger. If you have a few, display inline. Hiding just makes it worse to use, since it's harder to aim for it. And I personally dislike UI elements popping into existence as I move the cursor, but that might just be my personal grudge.

**Comment by ldexterldesign** [comment]
NO REPLY NECESSARY

Hi all,

Hope you're well

Thanks for software - just left Dashlane and it's wonderful; everything works!

I stumbled upon this issue while submitting another issue

> With regards to the copying fields by clicking, I'm not sure that this makes so much sense. For instance, I frequently copy only a portion of the field value. Having an auto-copy feature would hinder that and be less obvious from a UX perspective.

I agree, however an enhancement would be the ability to select/copy without having to enter the edit entry state - currently if I try to select (i.e. drag) it drags the whole window:

![untitled mov](https://user-images.githubusercontent.com/1715809/49249469-80aec880-f41c-11e8-8e39-09f9b12180fd.gif)

> I'm also not completely convinced that hiding the buttons is so bad in terms of UX. It hides clutter (which is bad UX) and you instantly learn about them when you hover one field. If they were all visible the interface would look horrible as well, imho.

I agree too, although I understand @darsain's concern with targeting buttons - perhaps keyboard navigation could be both a solution here and an overall enhancement

@darsain FYI https://en.wikipedia.org/wiki/Progressive_disclosure

If you have any issues (e.g. questions/queries) happy to help

Hope this helps!

Keep up great work!

Regards

**Comment by ldexterldesign** [comment]
FYI
NO REPLY NECESSARY

My todo app[p] employed this progressive disclosure technique a while back and seems to be a successful enhancement

![untitled mov](https://user-images.githubusercontent.com/1715809/50318689-0b865e80-04c3-11e9-8f99-e03bce7fd579.gif)

Hope this is useful!

Regards

[p] http://gqueues.com

**Comment by perry-mitchell** [comment]
Sorry everyone - I'm going to close this issue as it's going to be too hard to follow.

Please create separate issues if you wish to push any particular points forward. Feel free to copy content as it suits. Thanks!
